osu.Game/Beatmaps/Drawables/UpdateableBeatmapBackgroundSprite.cs

```diff
@@ -29,12 +32,13 @@ protected override Drawable CreateDrawable(BeatmapInfo model)
             {
                 Drawable drawable;
 
-                var localBeatmap = beatmaps.GetWorkingBeatmap(model);
-
-                if (localBeatmap.BeatmapInfo.ID == 0 && model?.BeatmapSet?.OnlineInfo != null)
-                    drawable = new BeatmapSetCover(model.BeatmapSet);
+                if (model?.BeatmapSet?.OnlineInfo != null)
+                    drawable = new BeatmapSetCover(model.BeatmapSet, beatmapSetCoverType);
                 else
-                    drawable = new BeatmapBackgroundSprite(localBeatmap);
+                {
+                    // Use the default background if somehow an online set does not exist.
```

Comment:
why not check local too, before falling back to default?

ie. why not just change the order of the two above rather than removing local fallback?

Reply:
was already in the middle of adding it back